overlord/servicestate/service_control.go

```diff
@@ -40,6 +40,7 @@ type ServiceAction struct {
 	Action         string   `json:"action"`
 	ActionModifier string   `json:"action-modifier,omitempty"`
 	Services       []string `json:"services,omitempty"`
+	ActionIsOnSnap bool     `json:"action-on-snap"`
```

Comment:
This goes in the state, so the less we have to write the better:

```suggestion
	ActionIsOnSnap bool     `json:"action-on-snap,omitempty"`
```

Reply:
I'm still not sure we need this flag per the discussion above.

Reply:
And it seems that in order to trigger the omitempty behaviour we would need to use a pointer to a bool: https://stackoverflow.com/questions/37756236/json-golang-boolean-omitempty

Reply:
A pointer would behave differently, as far as I understand the op had a different problem and needed a tri-state value. See https://play.golang.org/p/mbSupUbl5kv for pointer vs no-pointer. Also pointers make things more complicated wrt assignment.